aptos-move/framework/src/natives/signature.rs

```diff
@@ -123,6 +123,46 @@ pub fn native_ed25519_signature_verification(
     ))
 }
 
+pub fn native_ed25519_verify_proof_of_knowledge(
+    context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut arguments: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(_ty_args.is_empty());
+    debug_assert!(arguments.len() == 2);
+
+    // Verifying a proof-of-knowledge is more-or-less an Ed25519 signature verification, with a litte
+    // extra serialization cost.
+    let msg_len = ed25519::ED25519_PUBLIC_KEY_LENGTH * 2;
```

Comment:
Add a TODO here to tackle this with the upcoming gas schedule changes. We will have a special formula for this function.